packages/client/src/webStorage.ts

```diff
@@ -0,0 +1,261 @@
+            const [name, value] = cookie.trim().split("=");
+            return { name, value };
+        });
+
+        allOptionalStorage.forEach((storage) => {
+            const optionalStorageBase = storage.name.replace(/\*$/, "");
+            const matchedCookiesForDeletion = storedCookies.filter((cookie) =>
+                new RegExp(`^${optionalStorageBase}`, "i").test(cookie.name),
+            );
+
+            matchedCookiesForDeletion.forEach((cookie) => {
+                const domain = location.hostname.includes("nav.no")
+                    ? ".nav.no"
+                    : location.hostname;
+
+                Cookies.remove(cookie.name, {
+                    domain,
+                    path: "/",
+                    expires: 0,
+                });
+            });
+        });
+    }
+
+    private clearOptionalLocalAndSessionStorage(
+        allOptionalStorage: PublicStorageItem[],
+    ) {
+        const deleteStorage = (storage: Storage, name: string) => {
+            const optionalStorageBase = name.replace(/\*$/, "");
+            Object.keys(storage).forEach((key) => {
+                if (new RegExp(`^${optionalStorageBase}`, "i").test(key)) {
+                    storage.removeItem(key);
+                }
+            });
+        };
+        allOptionalStorage.forEach((storage) => {
+            deleteStorage(localStorage, storage.name);
+            deleteStorage(sessionStorage, storage.name);
+        });
+    }
+
+    private awaitDecoratorData = async () => {
+        return new Promise((resolve, reject) => {
+            const timeout = setTimeout(() => {
+                reject(
+                    new Error(
+                        `Timed out after ${DECORATOR_DATA_TIMEOUT}ms waiting for __DECORATOR_DATA__ to be set. Please check that the decorator is infact loading.`,
+                    ),
+                );
+            }, DECORATOR_DATA_TIMEOUT);
+
+            const checkForDecoratorData = () => {
+                if (window.__DECORATOR_DATA__) {
+                    clearTimeout(timeout);
+                    resolve(true);
+                } else {
+                    setTimeout(checkForDecoratorData, 50);
+                }
+            };
+
+            checkForDecoratorData();
+        });
+    };
+
+    private async clearOptionalStorage() {
+        await this.awaitDecoratorData();
+        const allowedStorage = this.getAllowedStorage();
+        const allOptionalStorage = allowedStorage.filter(
+            (storage) => storage.optional,
+        );
+
+        this.clearOptionalCookies(allOptionalStorage);
+        this.clearOptionalLocalAndSessionStorage(allOptionalStorage);
+    }
+
+    private checkAndTriggerConsentBanner() {
+        const { userActionTaken, meta } = this.getCurrentConsent();
+        const { version } = meta;
+
+        // Don't show cookie banner for nav.no editors
+        if (window.location.hostname.includes("oera.no")) {
```

Comment:
Ønsker det samme for arkiv
```suggestion
        if (window.location.hostname.includes("oera.no") || window.location.hostname.includes("cms-arkiv.ansatt") {
```